# Tarefa 03

- Leia os enunciados com atenção
- Saiba que pode haver mais de uma resposta correta
- Insira novas células de código sempre que achar necessário
- Em caso de dúvidas, procure os Tutores
- Divirta-se :)

In [19]:
import pandas as pd
import requests

####  1) Lendo de APIs
Vimos em aula como carregar dados públicos do governo através de um API (*Application Programming Interface*). No exemplo de aula, baixamos os dados de pedidos de verificação de limites (PVL) realizados por estados, e selecionamos apenas aqueles referentes ao estado de São Paulo.

1. Repita os mesmos passos feitos em aula, mas selecione os PVLs realizados por municípios no estado do Rio de Janeiro.
2. Quais são os três *status* das solicitações mais frequentes na base? Quais são suas frequências?
3. Construa uma nova variável que contenha o ano do **status**. Observe que ```data_status``` vem como tipo *object* no **DataFrame**. Dica: você pode usar o método ```.str``` para transformar o tipo da variável em string, em seguida um método como [**slice()**](https://pandas.pydata.org/docs/reference/api/pandas.Series.str.slice.html) ou [**split()**](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.str.split.html).
4. Indique a frequência de cada ano do campo construído no item (3).

In [20]:
# 1) Seu código aqui
url = 'https://apidatalake.tesouro.gov.br/ords/sadipem/tt/pvl?uf=RJ&tipo_interessado=Munic%C3%ADpio'
r = requests.get(url)
r.status_code

200

In [21]:
r.json()

{'items': [{'id_pleito': 5843,
   'tipo_interessado': 'Município',
   'interessado': 'Maricá',
   'cod_ibge': 3302700,
   'uf': 'RJ',
   'num_pvl': None,
   'status': 'Arquivado',
   'num_processo': '17944.000212/2010-10',
   'data_protocolo': '2010-11-30T02:00:00Z',
   'tipo_operacao': 'Operação contratual interna',
   'finalidade': 'PAC - Pró-Moradia - 2009',
   'tipo_credor': 'Instituição Financeira Nacional',
   'credor': 'Caixa Econômica Federal',
   'moeda': 'Real',
   'valor': 51016091.98,
   'pvl_assoc_divida': 0,
   'pvl_contradado_credor': 0,
   'data_status': '28/12/2010'},
  {'id_pleito': 4924,
   'tipo_interessado': 'Município',
   'interessado': 'Rio de Janeiro',
   'cod_ibge': 3304557,
   'uf': 'RJ',
   'num_pvl': None,
   'status': 'Arquivado',
   'num_processo': '10951.000224/2000-63',
   'data_protocolo': '2005-12-27T02:00:00Z',
   'tipo_operacao': 'Operação contratual externa (com garantia da União)',
   'finalidade': 'Preservação e recuperação ambiental',
   'tipo_c

In [22]:
data_json = r.json()
print(data_json)

{'items': [{'id_pleito': 5843, 'tipo_interessado': 'Município', 'interessado': 'Maricá', 'cod_ibge': 3302700, 'uf': 'RJ', 'num_pvl': None, 'status': 'Arquivado', 'num_processo': '17944.000212/2010-10', 'data_protocolo': '2010-11-30T02:00:00Z', 'tipo_operacao': 'Operação contratual interna', 'finalidade': 'PAC - Pró-Moradia - 2009', 'tipo_credor': 'Instituição Financeira Nacional', 'credor': 'Caixa Econômica Federal', 'moeda': 'Real', 'valor': 51016091.98, 'pvl_assoc_divida': 0, 'pvl_contradado_credor': 0, 'data_status': '28/12/2010'}, {'id_pleito': 4924, 'tipo_interessado': 'Município', 'interessado': 'Rio de Janeiro', 'cod_ibge': 3304557, 'uf': 'RJ', 'num_pvl': None, 'status': 'Arquivado', 'num_processo': '10951.000224/2000-63', 'data_protocolo': '2005-12-27T02:00:00Z', 'tipo_operacao': 'Operação contratual externa (com garantia da União)', 'finalidade': 'Preservação e recuperação ambiental', 'tipo_credor': 'Instituição Financeira Internacional', 'credor': 'Japan Bank for Internationa

In [23]:
print(data_json['items'])

[{'id_pleito': 5843, 'tipo_interessado': 'Município', 'interessado': 'Maricá', 'cod_ibge': 3302700, 'uf': 'RJ', 'num_pvl': None, 'status': 'Arquivado', 'num_processo': '17944.000212/2010-10', 'data_protocolo': '2010-11-30T02:00:00Z', 'tipo_operacao': 'Operação contratual interna', 'finalidade': 'PAC - Pró-Moradia - 2009', 'tipo_credor': 'Instituição Financeira Nacional', 'credor': 'Caixa Econômica Federal', 'moeda': 'Real', 'valor': 51016091.98, 'pvl_assoc_divida': 0, 'pvl_contradado_credor': 0, 'data_status': '28/12/2010'}, {'id_pleito': 4924, 'tipo_interessado': 'Município', 'interessado': 'Rio de Janeiro', 'cod_ibge': 3304557, 'uf': 'RJ', 'num_pvl': None, 'status': 'Arquivado', 'num_processo': '10951.000224/2000-63', 'data_protocolo': '2005-12-27T02:00:00Z', 'tipo_operacao': 'Operação contratual externa (com garantia da União)', 'finalidade': 'Preservação e recuperação ambiental', 'tipo_credor': 'Instituição Financeira Internacional', 'credor': 'Japan Bank for International Cooperat

In [24]:
Data_Rio = pd.DataFrame(data_json['items'])
Data_Rio.head()

id_pleito tipo_interessado         interessado  cod_ibge  uf num_pvl  \
0       5843        Município              Maricá   3302700  RJ    None   
1       4924        Município      Rio de Janeiro   3304557  RJ    None   
2       5864        Município         Duas Barras   3301603  RJ    None   
3       5328        Município  São João de Meriti   3305109  RJ    None   
4      11419        Município      Rio de Janeiro   3304557  RJ    None   

      status          num_processo        data_protocolo  \
0  Arquivado  17944.000212/2010-10  2010-11-30T02:00:00Z   
1  Arquivado  10951.000224/2000-63  2005-12-27T02:00:00Z   
2  Arquivado  17944.000218/2008-72  2008-04-14T03:00:00Z   
3  Arquivado  17944.000088/2007-97  2007-01-17T02:00:00Z   
4   Deferido  17944.001440/2008-92  2008-06-13T03:00:00Z   

                                       tipo_operacao  \
0                        Operação contratual interna   
1  Operação contratual externa (com garantia da U...   
2                        Operação contratual interna   
3                        Operação contratual interna   
4                        Operação contratual interna   

                            finalidade                           tipo_credor  \
0             PAC - Pró-Moradia - 2009       Instituição Financeira Nacional   
1  Preservação e recuperação ambiental  Instituição Financeira Internacional   
2                    Caminho da Escola       Instituição Financeira Nacional   
3                              Provias       Instituição Financeira Nacional   
4             PAC - Pró-Moradia - 2008       Instituição Financeira Nacional   

                                     credor moeda         valor  \
0                   Caixa Econômica Federal  Real  5.101609e+07   
1  Japan Bank for International Cooperation  Iene  1.836200e+07   
2                       Banco do Brasil S/A  Real  1.799000e+05   
3                       Banco do Brasil S/A  Real  1.980000e+06   
4                   Caixa Econômica Federal  Real  1.527289e+08   

   pvl_assoc_divida  pvl_contradado_credor data_status  
0                 0                      0  28/12/2010  
1                 0                      0  17/08/2006  
2                 0                      0  06/05/2008  
3                 0                      0  28/05/2007  
4                 1                      0  30/06/2008

In [25]:
# 2) Seu código aqui
Data_Rio.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 235 entries, 0 to 234
Data columns (total 18 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   id_pleito              235 non-null    int64  
 1   tipo_interessado       235 non-null    object 
 2   interessado            235 non-null    object 
 3   cod_ibge               235 non-null    int64  
 4   uf                     235 non-null    object 
 5   num_pvl                58 non-null     object 
 6   status                 235 non-null    object 
 7   num_processo           218 non-null    object 
 8   data_protocolo         218 non-null    object 
 9   tipo_operacao          235 non-null    object 
 10  finalidade             235 non-null    object 
 11  tipo_credor            235 non-null    object 
 12  credor                 235 non-null    object 
 13  moeda                  235 non-null    object 
 14  valor                  235 non-null    float64
 15  pvl_as

In [26]:
Data_Rio['status'].value_counts()

Deferido                                                 93
Arquivado                                                53
Regularizado                                             25
Arquivado por decurso de prazo                           23
Deferido (PVL-IF)                                        16
Encaminhado à PGFN com manifestação técnica favorável    13
Arquivado a pedido                                        6
Em retificação pelo interessado                           4
Em retificação pelo interessado (PVL-IF)                  1
Pendente de regularização                                 1
Name: status, dtype: int64

In [27]:
x = Data_Rio['interessado'].value_counts()
x

Rio de Janeiro           41
Resende                  15
Nova Iguaçu              14
Duque de Caxias          11
Niterói                   9
                         ..
Porciúncula               1
Rio das Flores            1
Varre-Sai                 1
Campos dos Goytacazes     1
Silva Jardim              1
Name: interessado, Length: 61, dtype: int64

In [29]:
y = solic_mais_frequente = (Data_Rio['interessado'] == 'Rio de Janeiro') | (Data_Rio['interessado'] == 'Resende') | (Data_Rio['interessado'] == 'Nova Iguaçu')
y

0      False
1       True
2      False
3      False
4       True
       ...  
230     True
231    False
232    False
233     True
234     True
Name: interessado, Length: 235, dtype: bool

In [30]:
Data_Rio[solic_mais_frequente]['status'].value_counts()

Deferido                                                 40
Arquivado                                                10
Deferido (PVL-IF)                                         6
Regularizado                                              5
Arquivado por decurso de prazo                            5
Encaminhado à PGFN com manifestação técnica favorável     4
Name: status, dtype: int64

In [31]:
# 3) Seu código aqui

data_status = Data_Rio['data_status']
data_status

0      28/12/2010
1      17/08/2006
2      06/05/2008
3      28/05/2007
4      30/06/2008
          ...    
230    25/05/2007
231    17/10/2019
232    22/10/2013
233    29/11/2018
234    13/07/2010
Name: data_status, Length: 235, dtype: object

In [32]:
ano = data_status.str.slice(6)
ano

0      2010
1      2006
2      2008
3      2007
4      2008
       ... 
230    2007
231    2019
232    2013
233    2018
234    2010
Name: data_status, Length: 235, dtype: object

In [33]:
ano.value_counts()

2008    37
2007    29
2014    20
2010    19
2019    17
2013    15
2020    15
2022    14
2012    13
2011    10
2009     9
2015     9
2021     5
2017     5
2006     4
2002     4
2018     4
2016     4
2004     1
2003     1
Name: data_status, dtype: int64

####  2) Melhorando a interação com o API
Observe dois URLs de consultas diferentes, por exemplo o URL utilizado em aula, e o URL feito no exercício anterior. Compare-os e observe as diferenças.

1. Faça uma função em Python que recebe como argumento o UF da consulta e o tipo de interessado (```'Estado'```ou ```Município```), e que devolve os dados da consulta no formato *DataFrame*.
2. Quantas solicitações para o Estado podem ser consultadas para Minas Gerais com *status* em 'Arquivado por decurso de prazo' estão registradas?
3. Qual é o município da Bahia com mais solicitações deferidas?
4. Salve um arquivo .csv com os dados de solicitações da Bahia, com interessado = 'Estado'

In [46]:
#1) Seu código aqui
def Data_UF (uf:str):
    url = 'https://apidatalake.tesouro.gov.br/ords/sadipem/tt/pvl?uf=UF&tipo_interessado=Munic%C3%ADpio'
    url = url.replace('UF',uf)
    r = requests.get(url)
    r.json()
    Data_json = r.json()
    Data_json = pd.DataFrame(Data_json['items'])

    return Data_json 

In [47]:
Data_UF ('GO')

id_pleito tipo_interessado interessado  cod_ibge  uf  \
0       19060           Estado       Goiás        52  GO   
1       24689           Estado       Goiás        52  GO   
2        7091           Estado       Goiás        52  GO   
3       12216           Estado       Goiás        52  GO   
4       10948           Estado       Goiás        52  GO   
5       24752           Estado       Goiás        52  GO   
6       10297           Estado       Goiás        52  GO   
7       13287           Estado       Goiás        52  GO   
8        7912           Estado       Goiás        52  GO   
9        5176           Estado       Goiás        52  GO   
10      24742           Estado       Goiás        52  GO   
11      10036           Estado       Goiás        52  GO   
12      13462           Estado       Goiás        52  GO   
13      11431           Estado       Goiás        52  GO   
14      13438           Estado       Goiás        52  GO   
15      18873           Estado       Goiás        52  GO   
16      11482           Estado       Goiás        52  GO   
17      10592           Estado       Goiás        52  GO   
18       6889           Estado       Goiás        52  GO   
19       9151           Estado       Goiás        52  GO   
20       7672           Estado       Goiás        52  GO   
21      12230           Estado       Goiás        52  GO   
22      13446           Estado       Goiás        52  GO   
23      22364           Estado       Goiás        52  GO   
24       4922           Estado       Goiás        52  GO   
25       9315           Estado       Goiás        52  GO   
26       9802           Estado       Goiás        52  GO   
27      10119           Estado       Goiás        52  GO   
28      27067           Estado       Goiás        52  GO   
29       7380           Estado       Goiás        52  GO   
30      11044           Estado       Goiás        52  GO   

                 num_pvl                                             status  \
0   00000.000000/1004-20                                 Arquivado a pedido   
1   PVL02.000675/2017-95                                 Arquivado a pedido   
2                   None                                           Deferido   
3                   None  Encaminhado à PGFN com manifestação técnica fa...   
4                   None                                          Arquivado   
5   PVL02.000790/2017-60                                           Deferido   
6                   None                     Arquivado por decurso de prazo   
7                   None  Encaminhado à PGFN com manifestação técnica fa...   
8                   None                                           Deferido   
9                   None                                          Arquivado   
10  PVL02.000741/2017-27                                 Arquivado a pedido   
11                  None  Encaminhado à PGFN com manifestação técnica fa...   
12                  None                                          Arquivado   
13                  None  Encaminhado à PGFN com manifestação técnica fa...   
14                  None                                          Arquivado   
15  00000.000000/2009-62  Encaminhado à PGFN com manifestação técnica fa...   
16                  None  Encaminhado à PGFN com manifestação técnica fa...   
17                  None  Encaminhado à PGFN com manifestação técnica fa...   
18                  None  Encaminhado à PGFN com manifestação técnica fa...   
19                  None                     Arquivado por decurso de prazo   
20                  None                     Arquivado por decurso de prazo   
21                  None  Encaminhado à PGFN com manifestação técnica fa...   
22                  None                                          Arquivado   
23  00000.000000/2029-42  Encaminhado à PGFN com manifestação técnica fa...   
24                  None                                           Deferido   
25                  None     

In [36]:
# 2) Seu código aqui
Data_UF ('MG')

id_pleito tipo_interessado      interessado  cod_ibge  uf  \
0          5507        Município  Pedra do Indaiá   3148905  MG   
1          5521        Município          Planura   3151602  MG   
2          5606        Município         Contagem   3118601  MG   
3          5613        Município          Camacho   3110400  MG   
4          5619        Município  Icaraí de Minas   3130051  MG   
...         ...              ...              ...       ...  ..   
4268      34066        Município        Arceburgo   3104106  MG   
4269      34074        Município           Arinos   3104502  MG   
4270       5491        Município       Abre Campo   3100302  MG   
4271      21653        Município          Guaxupé   3128709  MG   
4272       7951        Município         Argirita   3104403  MG   

                   num_pvl                          status  \
0                     None                        Deferido   
1                     None                        Deferido   
2                     None                        Deferido   
3                     None  Arquivado por decurso de prazo   
4                     None  Arquivado por decurso de prazo   
...                    ...                             ...   
4268  PVL02.003058/2019-11               Deferido (PVL-IF)   
4269  PVL02.003064/2019-61               Deferido (PVL-IF)   
4270                  None                        Deferido   
4271  00000.000000/2028-38                        Deferido   
4272                  None                        Deferido   

              num_processo        data_protocolo                tipo_operacao  \
0     17944.000128/2010-04  2010-04-15T03:00:00Z  Operação contratual interna   
1     17944.000132/2014-98  2014-04-07T03:00:00Z  Operação contratual interna   
2     17944.000153/2015-94  2015-04-06T03:00:00Z  Operação contratual interna   
3     17944.000155/2012-31  2012-03-29T03:00:00Z  Operação contratual interna   
4     17944.000156/2012-85  2012-04-27T03:00:00Z  Operação contratual interna   
...                    ...                   ...                          ...   
4268                  None                  None  Operação contratual interna   
4269                  None                  None  Operação contratual interna   
4270  17944.000124/2014-41  2014-05-13T03:00:00Z  Operação contratual interna   
4271  17944.000242/2016-11  2016-03-02T03:00:00Z  Operação contratual interna   
4272  17944.000683/2010-28  2010-07-02T03:00:00Z  Operação contratual interna   

                                             finalidade  \
0                              Caminho da Escola 2009 A   
1                                        Infraestrutura   
2     PAC 2 - Pró-Transporte - Pavimentação e Qualif...   
3                                          Provias 2010   
4                                        Infraestrutura   
...                                                 ...   
4268                                     Infraestrutura   
4269                                     Infraestrutura   
4270                                     Infraestrutura   
4271  Aquisição de terrenos, imóveis e/ou construção...   
4272     Aquisição de máquinas, equipamentos e veículos   

                          tipo_credor  \
0     Instituição Financeira Nacional   
1     Instituição Financeira Nacional   
2     Instituição Financeira Nacional   
3     Instituição Financeira Nacional   
4     Instituição Financeira Nacional   
...                               ...   
4268  Instituição Financeira Nacional   
4269  Instituição Financeira Nacional   
4270  Instituição Financeira Nacional   
4271  Instituição Financeira Nacional   
4272  Instituição Financeira Nacional   

                                            credor moeda        valor  \
0                              Banco do Brasil S/A  Real    185000.00   
1     Banco de Desenvolvimento de Minas Gerais S/A  Real   3000000.00   
2                          Caixa Econômica Federal  Real  1200

In [37]:
Data_MG = Data_UF ('MG')
s = (Data_MG['status'] == 'Arquivado por decurso de prazo')
Data_MG[s]['status'].count()

295

Podem ser feitas 295 consultas com o status "Arquivado por decurso de prazo"

In [38]:
# 3) Seu código aqui
Data_UF('BA')

id_pleito tipo_interessado             interessado  cod_ibge  uf  \
0         5954        Município  Oliveira dos Brejinhos   2923209  BA   
1         5972        Município                 Varzedo   2933174  BA   
2         6096        Município                  Mucuri   2922003  BA   
3         5015        Município              Alagoinhas   2900702  BA   
4         5692        Município          Cruz Das Almas   2909802  BA   
..         ...              ...                     ...       ...  ..   
876      18350        Município  Luís Eduardo Magalhães   2919553  BA   
877      31240        Município                 Boninal   2904001  BA   
878      18560        Município        Amélia Rodrigues   2901106  BA   
879      18580        Município    Conceição do Almeida   2908309  BA   
880      18590        Município               Itaparica   2916104  BA   

                  num_pvl                          status  \
0                    None                        Deferido   
1                    None  Arquivado por decurso de prazo   
2                    None                        Deferido   
3                    None                        Deferido   
4                    None                       Arquivado   
..                    ...                             ...   
876                  None                        Deferido   
877  PVL02.000545/2019-14               Deferido (PVL-IF)   
878                  None                        Deferido   
879                  None                       Arquivado   
880                  None                    Regularizado   

             num_processo        data_protocolo                tipo_operacao  \
0    17944.000243/2015-85  2015-03-04T03:00:00Z  Operação contratual interna   
1    17944.000247/2010-59  2010-03-17T03:00:00Z  Operação contratual interna   
2    17944.000276/2013-63  2013-08-30T03:00:00Z  Operação contratual interna   
3    17944.000024/2007-96  2007-01-09T02:00:00Z  Operação contratual interna   
4    17944.000174/2007-08  2007-01-26T02:00:00Z  Operação contratual interna   
..                    ...                   ...                          ...   
876  19407.000005/2002-68  2002-09-30T03:00:00Z  Operação contratual interna   
877                  None                  None  Operação contratual interna   
878  19407.000051/2003-48  2003-10-31T02:00:00Z  Operação contratual interna   
879  19407.000056/2003-71  2003-09-19T03:00:00Z  Operação contratual interna   
880  19407.000058/2006-11  2006-07-11T03:00:00Z  Operação contratual interna   

                                         finalidade  \
0                                    Infraestrutura   
1    Aquisição de máquinas, equipamentos e veículos   
2                                    Infraestrutura   
3                                           Provias   
4                                           Provias   
..                                              ...   
876  Aquisição de máquinas, equipamentos e veículos   
877                                  Infraestrutura   
878                                  Infraestrutura   
879                                            PMAT   
880         Regularização de Dívida - Água e Esgoto   

                         tipo_credor  \
0    Instituição Financeira Nacional   
1    Instituição Financeira Nacional   
2    Instituição Financeira Nacional   
3    Instituição Financeira Nacional   
4    Instituição Financeira Nacional   
..                               ...   
876  Instituição Financeira Nacional   
877  Instituição Financeira Nacional   
878  Instituição Financeira Nacional   
879  Instituição Financeira Nacional   
880                  Empresa Estatal   

                                        credor moeda       valor  \
0    Agência de Fomento do Estado da Bahia S/A  Real  4000000.00   
1    Agência de Fomento do Estado da Bahia S/A  Real    65000.00   
2    Agência de Fomento do Estado da Bahia S/A  Real  3800000.00   
3                   

In [45]:
Data_BA = Data_UF('BA')
z = (Data_BA['status'] == 'Deferido')
Data_BA[z]['interessado'].value_counts()

Luís Eduardo Magalhães    14
Vitória da Conquista      11
Lauro de Freitas          10
Camaçari                  10
Barreiras                  9
                          ..
Mansidão                   1
Nova Canaã                 1
Mutuípe                    1
Castro Alves               1
Amélia Rodrigues           1
Name: interessado, Length: 175, dtype: int64

O município de Luís Eduardo Magalhães teve mais solicitações deferidas no estado da Bahia.

In [54]:
# 4) Seu código aqui
url = 'https://apidatalake.tesouro.gov.br/ords/sadipem/tt/pvl?uf=BA&tipo_interessado=Estado'
r = requests.get(url)
r.json()
Data_json = r.json()
Data_json = pd.DataFrame(Data_json['items'])
Data_json

id_pleito tipo_interessado interessado  cod_ibge  uf num_pvl  \
0       10431           Estado       Bahia        29  BA    None   
1       18559           Estado       Bahia        29  BA    None   
2       18567           Estado       Bahia        29  BA    None   
3       18571           Estado       Bahia        29  BA    None   
4       18531           Estado       Bahia        29  BA    None   
..        ...              ...         ...       ...  ..     ...   
77      18356           Estado       Bahia        29  BA    None   
78      18551           Estado       Bahia        29  BA    None   
79      18575           Estado       Bahia        29  BA    None   
80      18579           Estado       Bahia        29  BA    None   
81      18587           Estado       Bahia        29  BA    None   

                                               status          num_processo  \
0   Encaminhado à PGFN com manifestação técnica fa...  17944.001198/2012-33   
1                                            Deferido  19407.000051/2002-67   
2                                            Deferido  19407.000053/2002-56   
3                                            Deferido  19407.000054/2002-09   
4                                            Deferido  19407.000044/2002-65   
..                                                ...                   ...   
77  Encaminhado à PGFN com manifestação técnica fa...  19407.000006/2003-93   
78                                           Deferido  19407.000049/2002-98   
79                                           Deferido  19407.000055/2002-45   
80                                           Deferido  19407.000056/2002-90   
81                                           Deferido  19407.000058/2002-89   

          data_protocolo                                      tipo_operacao  \
0   2013-07-04T03:00:00Z  Operação contratual externa (com garantia da U...   
1   2002-06-11T03:00:00Z                        Operação contratual interna   
2   2002-06-17T03:00:00Z                        Operação contratual interna   
3   2002-06-17T03:00:00Z                        Operação contratual interna   
4   2002-06-11T03:00:00Z                        Operação contratual interna   
..                   ...                                                ...   
77  2003-03-31T03:00:00Z  Operação contratual externa (com garantia da U...   
78  2002-06-21T03:00:00Z                        Operação contratual interna   
79  2002-06-17T03:00:00Z                        Operação contratual interna   
80  2002-06-19T03:00:00Z                        Operação contratual interna   
81  2002-06-17T03:00:00Z                        Operação contratual interna   

                     finalidade                           tipo_credor  \
0                Multissetorial  Instituição Financeira Internacional   
1             Saneamento básico       Instituição Financeira Nacional   
2             Saneamento básico       Instituição Financeira Nacional   
3                Infraestrutura       Instituição Financeira Nacional   
4             Saneamento básico       Instituição Financeira Nacional   
..                          ...                                   ...   
77  Desenvolvimento sustentável  Instituição Financeira Internacional   
78            Saneamento básico       Instituição Financeira Nacional   
79               Infraestrutura       Instituição Financeira Nacional   
80            Saneamento básico       Instituição Financeira Nacional   
81            Saneamento básico       Instituição Financeira Nacional   

                                               credor          moeda  \
0             Banco Interamericano de Desenvolvimento  Dólar dos EUA   
1                             Caixa Econômica Federal           Real   
2                             Caixa Econômica Federal           Real   
3                             Caixa Econômica Federal           Real   
4                             Caixa Econômica Federa

In [55]:
Data_json.to_csv('Dados_Bahia_Estado.csv')